inspirehep/modules/records/receivers.py

```diff
@@ -142,10 +145,11 @@ def push_to_orcid(sender, record, *args, **kwargs):
 
     task_name = current_app.config['ORCID_PUSH_TASK_ENDPOINT']
 
-    for orcid in get_orcids_for_push(record):
-        token = get_push_access_token(orcid)
-        if token is None:
-            continue
+    orcids = get_orcids_for_push(record)
+    remote_tokens = get_push_access_tokens(orcids)
+    for remote_token in remote_tokens:
+        access_token = remote_token.access_token
+        orcid = remote_token.remote_account.extra_data['orcid']
```

Comment:
doesn't it do N+1 queries now?  1 to get all tokens, N to get the corresponding orcid through this line.